### PR TITLE
Mulit-protocol app with protobufs

### PR DESCRIPTION
**Pull request by Stebalien**
Resubmission of @avive's PR (#249):

> Here's a little examples that shows how to build a node with multiple app protocols using protobufs as the wire format. The example creates a node that implements two protocols - ping and echo and the example demonstrate exchanging methods using the two protocols between two peers.

### CODE REVIEW

File: examples/multipro/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2014 Juan Batiz-Benet
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply examples/multipro/LICENSE]: (fixed)

**Comment by Stebalien** [comment]
@avive unfortunately, I believe it's a bit hard for me to give you write access to this branch. If you'd like to make further changes, just merge the PR I made against your repo and open a new PR.

Again, sorry about this.

**Comment by avive** [comment]
Ha - thanks for doing it. I got distracted by other open source tasks to be able to do it. Glad to contribute if this can somewhat assist in making the learning curve a bit less steep.

**Comment by vyzo** [comment]
go vet is failing the tests.

**Comment by vyzo** [comment]
fixed the generate protobuf.

**Comment by vyzo** [comment]
now we have some unfmt'ed files breaking the tests, will fix.

**Comment by avive** [comment]
I can fix anything. I've pulled to https://github.com/avive/go-libp2p but I have some deps/gx related problem building. Getting many error similar to this: 

> cannot find package "gx/ipfs/QmPgDWmTmuzvP7QE5zwo1TmjbJme9pmZHNujB2453jkCTr/go-libp2p-peerstore" in any of:
        /Users/auser/go/src/github.com/libp2p/go-libp2p/vendor/gx/ipfs/QmPgDWmTmuzvP7QE5zwo1TmjbJme9pmZHNujB2453jkCTr/go-libp2p-peerstore (vendor tree)

**Comment by avive** [comment]
I can issue a new PR once I can build and do some cleanups.

**Comment by vyzo** [comment]
you need to gx install first

**Comment by vyzo** [comment]
you can also issue a PR to this PR branch -- that's probably more effective for tracking the work.

**Comment by avive** [comment]
ok, created a new PR that should be good to go - https://github.com/libp2p/go-libp2p/pull/281

**Comment by vyzo** [comment]
the history has gotten a little messy, i think we should squash this.

**Comment by Stebalien** [comment]
> the history has gotten a little messy, i think we should squash this.

I'll look into that. I'd like to preserve some of the history at least.

**Comment by avive** [comment]
As a contributor - I'd also like not to get squashed :-)

**Comment by Stebalien** [comment]
> As a contributor - I'd also like not to get squashed :-)

I agree. We definitely want to preserve authorship. I've gotten rid of the back merge (and the duplicate commits, and the commits fixing up the broken merge).

**Pull request merged**